java/src/test/android/app/build.gradle

```
@@ -31,6 +40,25 @@ android {
 	kotlinOptions {
 		jvmTarget = '1.8'
 	}
+	// Conditional packagingOptions for QNN builds only
+	if (qnnVersion != null)	{
+		packagingOptions  {
+			jniLibs {
+				useLegacyPackaging = true
+			}
+			// Dsp is used in older QC devices and not supported by ORT
+			// Gpu support isn't the target, we just want Npu support (Htp)
+			exclude 'lib/arm64-v8a/libQnnGpu.so'
+			exclude 'lib/arm64-v8a/libQnnDsp*.so'
+		}
+	}
+	sourceSets {
+		main {
+			if (qnnVersion != null) {
+				manifest.srcFile 'src/main/AndroidManifestQnn.xml'  // Use QNN manifest
+			}
+		}
+	}
```

nit: we can remove one conditional

```suggestion
	if (qnnVersion != null)	{
		packagingOptions  {
			jniLibs {
				useLegacyPackaging = true
			}
			// Dsp is used in older QC devices and not supported by ORT
			// Gpu support isn't the target, we just want Npu support (Htp)
			exclude 'lib/arm64-v8a/libQnnGpu.so'
			exclude 'lib/arm64-v8a/libQnnDsp*.so'
		}
		sourceSets {
			main {
				manifest.srcFile 'src/main/AndroidManifestQnn.xml'  // Use QNN manifest
			}
		}
	}
```